### DISCUSSION TITLE
VSCode API function to compute line diffs between two texts

### DISCUSSION THREAD

**Discussion by BoykoAlex** [microsoft/vscode-discussions#1040] [Extension Development QnA] [answered]
I noticed that VSCode has functions able to compute line diffs between strings: `vs/editor/common/diff/linesDiffComputers`
However, I wasn't able to see line diffs functions in the public API that can be used in extensions... Is this correct?
Any chance you can expose any line diffs utilities as API?
(I know that i could bring in a 3rd party lib from npm for this. Someone on [SO](https://stackoverflow.com/questions/77897313/vscode-api-to-compute-line-differences-between-two-documents) mentioned that there is such API available...)

**Comment by abrilurena** [comment, marked as answer]
Hi, @BoykoAlex! I hope you're doing great. It's been a while, so I was wondering - do you still need help with this?
